tests/integration_test.rs

```diff
@@ -19,54 +19,106 @@ use predicates::prelude::*; // Used for writing assertions
 use std::process::Command; // Run programs
                            // use assert_cmd::cmd::Command; // Run programs - it seems to be equal to "use assert_cmd::prelude::* + use std::process::Command"
 
+use once_cell::sync::Lazy;
+use regex::bytes::Regex;
+use rusoto_core::Region;
+use rusoto_dynamodb::{DynamoDb, DynamoDbClient};
 use std::fs::File;
 use std::io::{self, Write}; // Used when check results by printing to stdout
+use std::sync::Mutex;
+use std::time::Duration;
+use tokio::time::delay_for;
 
 use tempfile::Builder;
 
 /// Integration tests would go with DynamoDB Local, so before running them setup() starts up DynamoDB Local with Docker.
 /// FYI: https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/DynamoDBLocal.html
 ///      https://hub.docker.com/r/amazon/dynamodb-local
-fn setup() -> Result</* std::process::Command */ Command, Box<dyn std::error::Error>> {
-    setup_with_port(8000)
+async fn setup() -> Result</* std::process::Command */ Command, Box<dyn std::error::Error>> {
+    setup_with_port(8000).await
 }
 
-fn setup_with_port(port: i32) -> Result<Command, Box<dyn std::error::Error>> {
-    // NOTE: setup() spins up DynamoDB Local in 8000 port as "local" region assumes it. Better to make it configurable.
-    let mut docker = Command::new("docker");
+// We use std::sync::Mutex instead of tokio::sync::Mutex, because mutex must be poisoned after setup failure.
+static SETUP_MUTEX: Lazy<Mutex<i32>> = Lazy::new(|| Mutex::new(0));
 
-    let check_cmd = docker.args(&[
+/// Check existence of docker process for dynamodb-local
+async fn check_dynamodb_local_running(port: u16) -> bool {
+    let mut docker_for_check = Command::new("docker");
+
+    let check_cmd = docker_for_check.args(&[
         "ps",
-        "-q",
-        "--filter",
-        &format!("expose={}", port),
+        "--format",
+        "{{.Ports}}",
         "--filter",
         "ancestor=amazon/dynamodb-local",
     ]);
-    let check_out = check_cmd.output().expect("failed to execut check cmd");
-    if check_out.stdout.len() != 0 {
-        println!("DynamoDB Local is already running.");
+    let check_out = check_cmd.output().expect("failed to execute check cmd");
+    let reg_str = format!(r"(?m)^[.\d]+:{}->\d+/tcp$", port);
```

Comment:
In the IPv6 available environment, this regex pattern doesn't match correctly.

Only IPv4 available environment:
```
$ docker ps --format {{.Ports}} --filter ancestor=amazon/dynamodb-local
0.0.0.0:8000->8000/tcp
```

IPv4 and IPv6 available environment:
```
$ docker ps --format {{.Ports}} --filter ancestor=amazon/dynamodb-local
0.0.0.0:8000->8000/tcp, :::8000->8000/tcp
```

Reply:
I fixed the regex and test on Ubuntu 21.10.